qt/scientific_interfaces/ISISReflectometry/GUI/Preview/PreviewPresenter.cpp

```diff
@@ -272,7 +271,33 @@ void PreviewPresenter::plotInstView() {
                                 m_instViewModel->getAxis());
 }
 
-void PreviewPresenter::plotRegionSelector() { m_regionSelector->updateWorkspace(m_model->getSummedWs()); }
+void PreviewPresenter::plotRegionSelector() {
+  if (!m_plotExistingROIs) {
+    updateRegionSelectorWorkspace();
+    return;
+  }
+
+  // If there are matching experiment settings already then add region selectors to
+  // display these on the plot
+  auto const roiMap = m_mainPresenter->getMatchingProcessingInstructionsForPreviewRow();
+  if (!roiMap.empty()) {
+    clearRegionSelector();
+  }
+
+  updateRegionSelectorWorkspace();
+
+  for (auto it = roiMap.cbegin(); it != roiMap.cend(); ++it) {
+    auto const &roiType = it->first;
+    auto const roiTypeString = roiTypeToString(roiType);
+    auto const color = roiTypeToColor(roiType);
+    auto const hatch = roiTypeToHatch(roiType);
+    auto const regions = Mantid::Kernel::Strings::parseGroups<size_t>(it->second);
+
+    for (auto const &region : regions) {
+      m_regionSelector->displayRectangularRegion(roiTypeString, color, hatch, region.front(), region.back());
```

Comment:
Are these not const refs because of requirements because of how python interface needs to handle them?

Reply:
I'm not sure about the python interface question I'm afraid, I'd need to give that try. I think what I wondered about these ones, though, is whether all of the methods are returning something that will go out of scope once the methods finish. I've just noticed now that the `roiTypeTo...` methods are inline, though, so perhaps that changes things? Or perhaps I'm misunderstanding something else about how references would work here?

Reply:
These ones seem to work fine as refs, but a bit of looking at [compiler explorer](https://godbolt.org/z/T9n8WGWzY) suggests that clang generally does a better job of optimising the copy out if it's left as-is. So apologies for the confusion there

Reply:
Ah, OK, good investigating, I'll change those back then.